src/tls_relation.py

```diff
@@ -207,9 +152,20 @@ def _get_private_key(self, hostname: str) -> Dict[str, str]:
         Returns:
             The encrypted private key.
         """
-        private_key_dict = {}
-        if JujuVersion.from_environ().has_secrets:
-            secret = self.charm_model.get_secret(label=f"private-key-{hostname}")
-            private_key_dict["key"] = secret.get_content()["key"]
-            private_key_dict["password"] = secret.get_content()["password"]
-        return private_key_dict
+        secret = self.model.get_secret(label=f"private-key-{hostname}")
+        private_key = secret.get_content()["key"]
+        password = secret.get_content()["password"]
+        return (private_key, password)
```

Comment:
I think you could create a `KeyPair` Namedtuple as this is often used at many places sometimes as a dict sometimes as a tuple

Reply:
Makes sense, I updated the method to return a `KeyPair` named tuple.